cpp/tests/interop/from_arrow_test.cpp

```diff
@@ -381,22 +380,20 @@ TEST_F(FromArrowTest, FixedPointTable)
   }
 }
 
-TEST_F(FromArrowTest, FixedPointTableLarge)
+TEST_F(FromArrowTest, FixedPoint128TableLarge)
 {
   using namespace numeric;
-  auto constexpr BIT_WIDTH_RATIO = 2;  // Array::Type:type::DECIMAL (128) / int64_t
-  auto constexpr NUM_ELEMENTS    = 1000;
+  auto constexpr NUM_ELEMENTS = 1000;
 
-  for (auto const i : {3, 2, 1, 0, -1, -2, -3}) {
-    auto every_other = [](auto i) { return i % BIT_WIDTH_RATIO ? 0 : i / BIT_WIDTH_RATIO; };
-    auto transform   = cudf::detail::make_counting_transform_iterator(BIT_WIDTH_RATIO, every_other);
-    auto const data  = std::vector<int64_t>(transform, transform + NUM_ELEMENTS * BIT_WIDTH_RATIO);
-    auto iota        = thrust::make_counting_iterator(1);
-    auto const col   = fp_wrapper<int64_t>(iota, iota + NUM_ELEMENTS, scale_type{i});
+  for (auto const scale : {3, 2, 1, 0, -1, -2, -3}) {
+    auto iota           = thrust::make_counting_iterator(1);
+    auto const data     = std::vector<__int128_t>(iota, iota + NUM_ELEMENTS);
+    auto const col      = fp_wrapper<__int128_t>(iota, iota + NUM_ELEMENTS, scale_type{scale});
     auto const expected = cudf::table_view({col});
 
     std::shared_ptr<arrow::Array> arr;
-    arrow::Decimal128Builder decimal_builder(arrow::decimal(10, -i), arrow::default_memory_pool());
+    arrow::Decimal128Builder decimal_builder(arrow::decimal(10, -scale),
+                                             arrow::default_memory_pool());
     decimal_builder.AppendValues(reinterpret_cast<const uint8_t*>(data.data()), NUM_ELEMENTS);
     CUDF_EXPECTS(decimal_builder.Finish(&arr).ok(), "Failed to build array");
```

Comment:
These 5 lines can be replaced with `auto const arr = make_decimal128_arrow_array(data, std::nullopt, scale);`